tools/utils/setup-environment.ts

```diff
@@ -5,17 +5,25 @@ import * as log from 'npmlog';
 import * as dotenv from 'dotenv';
 import type { ExecutorContext } from '@nrwl/devkit';
 
-process.env['NX_GIT_COMMIT_HASH'] = execSync('git rev-parse HEAD')
-  .toString()
-  .replace(/[\r\n]/gm, '');
-process.env['NX_GIT_BRANCH'] = execSync('git rev-parse --abbrev-ref HEAD')
-  .toString()
-  .replace(/[\r\n]/gm, '');
-process.env['NX_GIT_ORIGIN_URL'] = execSync('git remote get-url origin')
-  .toString()
-  .replace('ssh://git@', 'https://')
-  .replace('.git', '')
-  .replace(/[\r\n]/gm, '');
+try {
+  process.env['NX_GIT_COMMIT_HASH'] =
+    process.env['COMMIT_REF'] ??
+    execSync('git rev-parse HEAD')
```

Comment:
chore (non-blocking): if these are only going to be used in a dev environement I would be ok with just simplifying this and setting hash/branch/repo to just "dev" rather than scripts